packages/node-core/src/indexer/storeCache/cachePoi.ts

```diff
@@ -74,8 +74,18 @@ export class CachePoiModel extends Cacheable implements ICachedModelControl, Poi
     ]);
   }
 
-  protected clear(): void {
-    this.setCache = {};
-    this.flushableRecordCounter = 0;
+  clear(blockHeight?: number): void {
+    if (!blockHeight) {
+      this.setCache = {};
+      this.flushableRecordCounter = 0;
+      return;
+    }
+    // Clear everything below the block height
+    for (const [n, p] of Object.entries(this.setCache)) {
+      if (Number(n) <= blockHeight) {
+        delete this.setCache[Number(n)];
```

Comment:
Use copy instead, mutate the array that is currently looping will cause data corruption